R/plot_heatmap.R

```diff
@@ -0,0 +1,143 @@
+#' @param show_column_names Boolean check. If \code{True}, column names are shown. Otherwise, they aren't. 
+#' Default \code{FALSE}
+#' @param show_row_names Boolean check. If \code{True}, row names are shown. Otherwise, they aren't. 
+#' Default \code{FALSE}
+#' @param scale Boolean check. If \code{True}, values are scaled by z-score. Otherwise, they aren't.
+#' Default \code{TRUE}
+#' @param annotation Users have the option of plotting the exposure matrix based on their given annotation like
+#' Tumor_Subtypes or age. Error given if the user given annotation doesn't exist in the result annotation object.
+#' @param subset_tumor Users can specify certain tumor types on which they want to subset the exposure matrix for 
+#' plotting the heatmap. 
+#' @param subset_signatures Users can specify certain signatures on which they want to subset the exposure matrix 
+#' plotting the heatmap.
+#' @return Generates a heatmap for using the exposure matrix.
+#' @examples result <- readRDS(system.file("testdata", "res_annot.rds",
+#' package = "musicatk"))
+#' exp <- exposures(result)
+#' ComplexHeatmap::Heatmap(exp)
+#' @export
+
+
+
+plot_heatmap <- function(result, 
+                         proportional = FALSE,
+                         show_column_names = FALSE,
+                         show_row_names = TRUE,
+                         scale = TRUE,
+                         subset_tumor = NULL,
+                         subset_signatures = NULL,
+                         annotation = NULL,
+                         ...
+                         ){
+  
+  exp <- exposures(result) #extracting the exposures matrix
+  
+  if(isTRUE(proportional)){
+    exp <- sweep(exp, 2, colSums(exp), FUN = "/") #normalizing 
+  }
+  
+  if(isTRUE(scale)){
+    exp <- scale(exp)
+  }
+  
+  
+  annot <- sample_annotations(result) #Extracting annotations from musicatk object
+  ha = NULL #initilaizing empty annotation object
+  ha1 = NULL
+  
+   #checking if annotation argument correct
+  if(!is.null(annotation)){
+    if(!annotation %in% colnames(annot)) { 
+      stop("The given annotations are not present in the data") 
+    }
+    annot <- annot[,..annotation]
+    ha<- ComplexHeatmap::HeatmapAnnotation(df = annot)
+  }
+  
+  if(!is.null(subset_signatures)){
+    if(!subset_signatures %in% rownames(exp)) { 
+      stop("The given signatures are not present in the data") 
+    }
+    exp <- as.data.frame(exp)
+    exp <- subset(exp,rownames(exp) %in% subset_signatures)
+    exp <- as.matrix(exp)
+  }
+  
+  if(!is.null(subset_tumor)){
+    annot <- sample_annotations(result)
+    samps <-  subset(annot, Tumor_Subtypes == subset_tumor)
+    samps <- as.character(samps$Samples)
+    
+    exp <- as.data.frame(exp)
+    exp <- dplyr::select(exp,samps) #Selecting columns that match tumor subtype
+    exp <- as.matrix(exp)
+    
+    annot <- annot$Tumor_Subtypes
+    annot <- annot[annot == subset_tumor]
+    ha <- ComplexHeatmap::HeatmapAnnotation(df = annot)
```

Comment:
If there's an annotation given and a subset is specified does the heatmap run twice? If so find a way to refactor the code so that HeatmapAnnotation can be called just once.

Reply:
Would it be logical to add a condition to check if both the annotation and subset_tumor parameters are passed by the user and if the annotation is "Tumor_Subtypes" to just run the HeatmapAnnotation chunk for the subset_tumor?

Reply:
I think you can just move the HeatmapAnnotation call out of the logical blocks entirely. It needs to be called no matter what, and subsetting on exposures or signatures will just modify the input

Reply:
Yeah, that makes sense. I placed it outside of the logical blocks at the end in the latest commit.

Reply:
Nevermind. I think I misunderstand how the heatmap works. The "heatmap" variable is an optional input to Heatmap()? If so I think you may be okay the way you are.

Reply:
yeah, it's optional. Should I move it back into the logical blocks like I had it earlier?

Reply:
I am assuming you meant to keep it the way I had it so I have put it back in the logical blocks like I had earlier in the latest commit. Let me know if that's not what you wanted.